ambry-store/src/main/java/com/github/ambry/store/StorageManager.java

```diff
@@ -512,35 +512,45 @@ public void onPartitionBecomeInactiveFromStandby(String partitionName) {
       // if replica is null that means partition is not on current node (this shouldn't happen unless we use server admin
       // tool to remove the store before initiating decommission on this partition). We throw exception in this case.
       if (replica != null) {
-        // 0. as long as local replica exists, we create a decommission file in its dir
-        File decommissionFile = new File(replica.getReplicaPath(), BlobStore.DECOMMISSION_FILE_NAME);
-        try {
-          if (!decommissionFile.exists()) {
-            // if not present, create one.
-            decommissionFile.createNewFile();
-            logger.info("Decommission file is created for replica {}", replica.getReplicaPath());
-          }
-        } catch (IOException e) {
-          logger.error("IOException occurs when creating decommission file for replica " + partitionName, e);
-          throw new StateTransitionException(
-              "Couldn't create decommission file for replica " + replica.getReplicaPath(), ReplicaOperationFailure);
+        Store localStore = getStore(replica.getPartitionId(), true);
+        if (localStore == null) {
+          throw new StateTransitionException("Store " + partitionName + " is not found on current node",
+              ReplicaNotFound);
         }
-        Store localStore = getStore(replica.getPartitionId());
-        if (localStore != null) {
-          // 1. set state to INACTIVE
-          localStore.setCurrentState(ReplicaState.INACTIVE);
-          logger.info("Store {} is set to INACTIVE", partitionName);
-          // 2. disable compaction on this store
-          if (!controlCompactionForBlobStore(replica.getPartitionId(), false)) {
-            logger.error("Failed to disable compaction on store {}", partitionName);
-            // we set error code to ReplicaNotFound because that is the only reason why compaction may fail.
-            throw new StateTransitionException("Couldn't disable compaction on replica " + replica.getReplicaPath(),
-                ReplicaNotFound);
-          }
-          logger.info("Compaction is successfully disabled on store {}", partitionName);
+        if (localStore.disabledOnError()) {
+          // if store is disabled due to disk I/O error, we explicitly throw an exception to mark partition as ERROR state
+          throw new StateTransitionException("Store " + partitionName + " is disabled due to I/O error",
+              ReplicaOperationFailure);
         } else {
```

Comment:
minor: since the added if throws, this code doesn't need to be in an else clause. This can shorten the diff a bit and seems to match the existing style of this method a bit better.

Reply:
Touche!